ProjFS.Mac/PrjFSKextTests/KextMockUtilities.hpp

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <unordered_map>
+#include <unordered_set>
+
+class FunctionTypeCallRecorderBase
```

Comment:
I struggled a bit with naming the classes and functions here, due to my limited exposure to extensive automated testing. I suspect there is existing terminology I should be applying here.

The idea is to be able to check whether specific mock functions were called or not by recording mock function calls in a central registry that can be interrogated after the test code has run. It currently doesn't support it, but the idea is to store the argument values too. ("Was function X called with an argument Y?") It already records calls with a sequence number; right now that number isn't used, but the idea is to be able to verify that e.g. a test calls F1() before F2().

Reply:
@Pmj if you're looking for inspiration [moq](https://github.com/moq/moq4) is a very popular c# testing framework.

I can understand what you've done so far.  I think it can grow organically over time as needed.

Reply:
> if you're looking for inspiration [moq](https://github.com/moq/moq4) is a very popular c# testing framework.

Thanks, I'll check it out!

Reply:
Thanks, I've incorporated those naming suggestions now.